include/uprotocol-cpp/uri/tools/Utils.h

```diff
@@ -0,0 +1,186 @@
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * 
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ 
+ 
+ */
+
+#ifndef UPROTOCOL_CPP_UTILS_H
+#define UPROTOCOL_CPP_UTILS_H
+
+#include <algorithm>
+#include <cctype>
+#include <optional>
+#include <string>
+#include <array>
+#include <string_view>
+#include <arpa/inet.h>
+#include <spdlog/spdlog.h>
+#include "uprotocol/uri.pb.h"
+
+namespace uprotocol::uri {
+
+/**
+ *  
+ * @param str 
+ * @return 
+ */
+    [[nodiscard]] [[maybe_unused]] static auto isBlank(std::string_view str) -> bool {
+        return std::all_of(str.begin(), str.end(), [](char ch) { return std::isspace(ch); });
+    }
+
+/**
+ * 
+ * @param entity 
+ * @return 
+ */
+    [[nodiscard]] [[maybe_unused]] static auto isEmpty(uprotocol::v1::UEntity const &entity) -> bool {
+        return entity.name().empty() && !entity.has_version_major() && !entity.has_id();
+    }
+
+/**
+ * 
+ * @param resource 
+ * @return 
+ */
+    [[nodiscard]] [[maybe_unused]] static auto isEmpty(uprotocol::v1::UResource const &resource) -> bool {
+        auto const &name = resource.name();
+        return (name.empty() || "rpc" == name) &&
+                (!resource.has_instance() || resource.instance().empty()) &&
+                (!resource.has_message() || resource.message().empty()) &&
+                (!resource.has_id() || resource.id() == 0);
+    }
+
+
+/**
+ * 
+ * @param authority 
+ * @return 
+ */
+    [[nodiscard]] [[maybe_unused]] static auto isEmpty(uprotocol::v1::UAuthority const &authority) -> bool {
+        return !((authority.has_name() && !authority.name().empty()) ||
+            (authority.has_id() && !authority.id().empty()) ||
+            (authority.has_ip() && !authority.ip().empty()));
+    }
+    
+    [[nodiscard]] [[maybe_unused]] static auto isEmpty(uprotocol::v1::UUri const &uri) -> bool {
+        return isEmpty(uri.authority()) && isEmpty(uri.resource()) && isEmpty(uri.entity());
+    }
+    
+    
+    [[nodiscard]] [[maybe_unused]] auto static isResolved([[maybe_unused]] uprotocol::v1::UUri const &uri) -> bool {
+        return false;
```

Comment:
Are these not implemented yet? If so please remove them or comment them out so people don't use them.